db/db_iterator_test.cc

```diff
@@ -3555,6 +3555,54 @@ TEST_F(DBIteratorTest, ErrorWhenReadFile) {
   iter->Reset();
 }
 
+TEST_F(DBIteratorTest, MultipleIteratorsConsistentViewTest) {
+  // TODO - This test is to demonstrate that NewIterators() API
+  // currently does not return consistent view of the database across multiple
+  // column families. Modify this test to check the consistency when
+  // NewIterators API is fixed
```

Comment:
I'm wondering if it would make sense to just commit the (updated) test alongside the fix? I mean, the problem with the current `NewIterators` is fairly obvious; the thing that surprised me the most here is that we apparently claim that it provides a consistent view when it clearly does not (unless the application passes in its own snapshot).

Reply:
If we actually wanted to demonstrate that the view is not guaranteed to be consistent, we would have to show that KVs necessary for a certain snapshot can be lost during a concurrent flush. Not sure if it's worth the effort if we're about to fix this.

Reply:
I was debating myself whether I should just fix it or this PR is worth merging in anyway.

Let me put up a new PR to fix the issue. I can just close this without merging and link in the new PR.